skbio/sequence/_sequence.py

```diff
@@ -187,30 +351,107 @@ def metadata(self):
     def metadata(self, metadata):
         if not isinstance(metadata, dict):
             raise TypeError("metadata must be a dict")
-        self._metadata = metadata
+        # shallow copy
+        self._metadata = metadata.copy()
+
+    @metadata.deleter
+    def metadata(self):
+        if hasattr(self, '_metadata'):
```

Comment:
With efficient handling of missing metadata this code was refactored. It's much clearer now, both here and in the constructor.